examples/partial_labels.py

```diff
@@ -0,0 +1,182 @@
+    )
+    parser.add_argument(
+        "--dev_dataset_id",
+        required=True,
+        help="Dataset ID for dev dataset associated w/ prod workflow",
+    )
+    parser.add_argument(
+        "--dev_questionnaire_id",
+        required=True,
+        help="Questionnaire (teach task) ID associated with dev dataset",
+    )
+    parser.add_argument(
+        "--dev_model_group_id",
+        required=True,
+        help="Model Group ID for dev dataset / teach task",
+    )
+    parser.add_argument(
+        "--dev_workflow_id",
+        required=True,
+        help="Workflow ID for dev workflow associated with teach task",
+    )
+    parser.add_argument(
+        "--prod_workflow_id",
+        required=True,
+        help="Workflow ID for prod workflow with review data",
+    )
+    parser.add_argument(
+        "--max_review_docs",
+        default=1000,
+        help="Maximum number of documents from review to add to dataset",
+    )
+    parser.add_argument(
+        "--oversample_errs",
+        action="store_true",
+        help="Sample more heavily from documents with more post review corrections",
+    )
+    return parser
+
+
+def main(
+    dev_host: str,
+    dev_api_token_path: str,
+    prod_host: str,
+    prod_api_token_path: str,
+    dev_dataset_id: int,
+    dev_questionnaire_id: int,
+    dev_model_group_id: int,
+    dev_workflow_id: int,
+    prod_workflow_id: int,
+    max_review_docs: int = 1000,
+    oversample_errs: bool = False,
+):
+    """
+    End to end function to data that has been submitted to a production workflow
+    and reviewed to an existing dataset, teach task, and extraction model
+
+    TODO
+     * Use max_review_docs and oversample_errs arguments
+
+    Args:
+        dev_host: Host URL for dev environment
+        dev_api_token_path: API token path for dev env
+        prod_host: Host URL for prod env
+        prod_api_token_path: API token path for prod env
+        dev_dataset_id: Dataset ID for dev dataset associated w/ prod workflow
+        dev_questionnaire_id: Questionnaire (teach task) ID associated with dev dataset.
+            Necessary because a dataset can have multiple teach tasks
+        dev_model_group_id: Model Group ID for dev dataset / teach task
+        dev_workflow_id: Workflow ID for dev workflow. Required for adding data
+            to teach task and model
+        prod_workflow_id: Workflow ID for prod workflow w/ review enabled
+        max_review_docs: Maximum number of documents from review to add to dataset
+            If max_review_docs is less than the number of docs that have been
+            reviewed, sample from review
+        oversample_errs: If True, sample more heavily from documents with more
+            discrepancies between model predictions and ground truth
+    """
+    # Connect to dev
+    dev_client = IndicoClient(
+        IndicoConfig(host=dev_host, api_token_path=dev_api_token_path)
```

Comment:
I wonder if this should also be a class where we pass the instantiated client in __init__, that way people can connect with config kwargs (e.g. verify_ssl=False is not uncommon w/ users). This is also more in line with most other classes

Reply:
If this is being run from the command line (how I have it set up) I'm not sure how to best pass in an instantiated client. But I do see your point - I could modify so that at least the main function signature takes in clients, so that method could be called directly rather than using command line args